### PR TITLE
Rework exception handling

### PR DESCRIPTION
**Pull request by espkk**
- Rework exception handling
- Improve core dumps (add exception info on assert())
- Manage the lifetime of CDebug

Addresses #55

### CODE REVIEW

**Comment by espkk** [comment]
@np-c0mp1ete wtf is that clang-format check hook? it does not accept code formatted with the latest clang-format

**Comment by np-c0mp1ete** [comment]
> @np-c0mp1ete wtf is that clang-format check hook? it does not accept code formatted with the latest clang-format

I believe that's because you have to use the one that comes with the latest VS and that is clang-format version 11.0.0.

**Comment by espkk** [comment]
> 
> 
> > @np-c0mp1ete wtf is that clang-format check hook? it does not accept code formatted with the latest clang-format
> 
> I believe that's because you have to use the one that comes with the latest VS and that is clang-format version 11.0.0.

well... idk how to format all the files with internal clang-format. could you help here?

**Comment by np-c0mp1ete** [comment]
> > > @np-c0mp1ete wtf is that clang-format check hook? it does not accept code formatted with the latest clang-format
> > 
> > 
> > I believe that's because you have to use the one that comes with the latest VS and that is clang-format version 11.0.0.
> 
> well... idk how to format all the files with internal clang-format. could you help here?

Try `for /R %f in (*.cpp *.h *.hpp) do clang-format -i "%f"` from VS Command Prompt. This command assumes that you are in the src folder

File: src/apps/ENGINE/CMakeLists.txt [hunk 1]
```diff
@@ -2,6 +2,10 @@ project(engine)
 
 find_package(Fmod REQUIRED) # required to set ${Fmod_LIBRARY_PATHS}
 
+if(WIN32)
+    add_definitions("/EHa")
```
Comment (np-c0mp1ete) [inline src/apps/ENGINE/CMakeLists.txt:6]: - Why not `add_compile_options()`?
- Isn't it MSVC-specific? I'd suggest putting it in https://github.com/storm-devs/storm-engine/blob/develop/src/CMakeLists.txt next to other compiler-specific options

**Review by np-c0mp1ete — CHANGES_REQUESTED** [review]
Naming conventions in exceptions.hpp don't follow the ones from the Style Guide PR and the rest of the codebase in general.

**Comment by espkk** [comment]
switched to https://github.com/storm-devs/storm-engine/pull/58 solution for now

**Pull request closed without merging**